src/main/java/net/consensys/orion/enclave/sodium/SodiumEnclave.java

```diff
@@ -64,7 +65,12 @@ public EncryptedPayload encrypt(byte[] plaintext, Box.PublicKey senderKey, Box.P
     final EncryptedKey[] encryptedKeys =
         encryptPayloadKeyForRecipients(payloadKey, recipientsAndSender, senderSecretKey, nonce);
 
-    final byte[] privacyGroupId = generatePrivacyGroupId(recipientsAndSender);
+    final byte[] privacyGroupId;
+    if (seed == null) {
```

Comment:
nit: Consider moving this logic down into generatePrivacyGroupId()

Reply:
Agreed